src/main/java/hng_java_boilerplate/config/SecurityConfig.java

```diff
@@ -0,0 +1,38 @@
+package hng_java_boilerplate.config;
+
+import hng_java_boilerplate.jwt.JwtAuthFilter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig {
+
+    @Autowired
+    private AuthenticationProvider authenticationProvider;
+    @Autowired
+    private JwtAuthFilter jwtAuthFilter;
```

Comment:
```suggestion
    private  final JwtAuthFilter jwtAuthFilter;
```
Add @RequiredArgsConstructor

Make both final